drivers/include/abp2.h

```diff
@@ -0,0 +1,283 @@
+
+#include "saul.h"
+
+#if defined(MODULE_ABP2_SPI)
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#elif defined(MODULE_ABP2_I2C)
+#include "periph/i2c.h"
+#else
+#error "ABP2 driver >> Please add a valid module: either abp2_spi or abp2_i2c"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(MODULE_ABP2_I2C)
+/**
+ * @brief   Default I2C slave address for ABP2 devices
+ *
+ * The sensor address cannot be changed.
+ * A different address can only be obtained upon special request to Honeywell.
+ */
+#define ABP2_ADDR_DEFAULT       (0x28)
+#endif
+
+
+/**
+ * @defgroup abp2_statusbits Status bits
+ * @{
+ *  Macro definitions to manipulate the bits of the status byte.
+ *
+ * Status byte explanation :
+ *
+ *  Bit | Meaning                       | Comment |
+ *  --- | ---                           | --- |
+ *  7   | always 0                      | |
+ *  6   | 1 : device is powered         | |
+ *  5   | 1 : device is busy            | Wait longer between 2 commands. |
+ *  4   | always 0                      | |
+ *  3   | always 0                      | |
+ *  2   | 1 : integrity test failed     | Checksum error at power-up. |
+ *  1   | always 0                      | |
+ *  0   | 1 : internal math saturation  | |
+ */
+#define ABP2_STATUS_BUSY        0x20    /**< Sensor is busy. */
+#define ABP2_STATUS_MEMERR      0x04    /**< Memory integrity error. */
+#define ABP2_STATUS_MATHSAT     0x01    /**< Math saturation error. */
+/** A bitwise AND of the status byte with this mask will return true in case of error. */
+#define ABP2_STATUS_MASK        (ABP2_STATUS_BUSY | ABP2_STATUS_MEMERR | ABP2_STATUS_MATHSAT)
+/** @} */ /* end of abp2_statusbits */
+
+/**
+ * @brief   Parameters for ABP2 sensors.
+ */
+typedef struct abp2_params {
+#if defined(MODULE_ABP2_SPI)
+    spi_t spi;              /**< SPI bus the sensor is connected to. */
+    spi_cs_t cs;            /**< CS pin GPIO handle. */
+    spi_clk_t clk;          /**< SPI bus clock speed. */
+#else
+    i2c_t i2c;              /**< I2C bus the device is connected to. */
+    uint8_t addr;           /**< Device address on the I2C bus. */
+#endif
+    float rangeMin;         /**< Minimum pressure range (sensor unit). */
+    float rangeMax;         /**< Maximum pressure range (sensor unit). */
+} abp2_params_t;
+
+/**
+ * @brief   Device descriptor for ABP2 sensors.
+ */
+typedef struct abp2 {
+    const abp2_params_t *params;   /**< Sensor parameters. */
+} abp2_t;
+
+/**
+ * @brief Raw values read from a ABP2 sensor.
+ */
+typedef struct abp2_raw {
+    uint32_t cntPress;      /**< 24-bit digital pressure reading (counts). */
+    uint32_t cntTemp;       /**< 24-bit digital temperature reading (counts). */
+} abp2_raw_t;
+
+
+/**
+ * @brief ABP2 error codes.
+ */
+enum {
+    ABP2_OK         =  0,       /**< No error. */
+    ABP2_ERRBUS     = -1,       /**< SPI/I2C communication failed. */
+    ABP2_ERRSENS    = -2        /**< Sensor error, see status byte. */
```

Comment:
I'd suggest to just use values from `errno.h` instead of defining custom error codes.

Reply:
OK.